gulpfile.js

```diff
@@ -0,0 +1,199 @@
+const _getBabelConfig = format => ({
+  babelrc: false,
+
+  presets: [
+    format === FORMAT_ESM ? ESM_ENV_PRESET : "@babel/env",
+    "@babel/typescript"
+  ],
+  plugins: [
+    "@babel/proposal-class-properties",
+    "@babel/proposal-object-rest-spread",
+    "@babel/plugin-external-helpers",
+    ...(format === FORMAT_UMD ? [UMD_TRANSFORM_PLUGIN] : [])
+  ]
+});
+
+const _getBabelStream = format =>
+  gulp
+    // get a stream of the files to transpile
+    .src(FILES_TO_BUILD)
+    // initialize the sourcemaps (used by UMD only)
+    .pipe(format === FORMAT_UMD ? sourcemaps.init() : util.noop())
+    // do the appropriate babel transpile (this is a copy from package.json)
+    .pipe(babel(_getBabelConfig(format)));
+
+const _genUmd = ({ minify = false } = {}) =>
+  _getBabelStream(FORMAT_UMD)
+    // If you're using UMD, you probably don't have `process.env.NODE_ENV` so, we'll replace it.
+    // If you're using the unminified UMD, you're probably in DEV
+    // If you're using the unminified UMD, you're probably in production
+    .pipe(
+      replace(
+        "process.env.NODE_ENV",
+        JSON.stringify(minify ? "production" : "development")
+      )
+    )
+    // minify the files and rename to .min.js extension (when minifying)
+    .pipe(minify ? uglify() : util.noop())
+    .pipe(minify ? rename({ extname: ".min.js" }) : util.noop())
+    .pipe(sourcemaps.write("./"))
+    .pipe(
+      debug({
+        title: minify ? "Building + Minifying UMD:" : "Building UMD:"
+      })
+    )
+    .pipe(gulp.dest("lib/umd"));
+
+const _genDist = ({ minify = false } = {}) =>
+  rollup({
+    input: SOURCE_ENTRY,
+
+    plugins: [
+      // Need to replace `process.env.NODE_ENV` in the bundle because most likely the place where this
+      // would be used doesn't support it. When minified we assume production, dev otherwise
+      rollupReplace({
+        "process.env.NODE_ENV": JSON.stringify(
+          minify ? "production" : "development"
+        )
+      }),
+
+      // convert JSON files to ES6 modules, so they can be included in Rollup bundle
+      rollupJson(),
+
+      // Locate modules using the Node resolution algorithm, for using third party modules in node_modules
+      rollupNodeResolve({
+        // use "module" field for ES6 module if possible
+        module: true,
+
+        // use (legacy) "jsnext:main" if possible
+        jsnext: true,
+
+        // use "main" field or index.(ts|js)
+        main: true,
+
+        // use "browser" field if possible
+        browser: true,
+
+        // include typescript files as default extensions
+        extensions: [".ts", ".js"]
+      }),
+
+      // Convert CommonJS modules to ES6 modules, so they can be included in a Rollup bundle
+      rollupCommonjs({
+        // Node modules are the ones we're trying to get it to understand
+        include: "node_modules/**"
+      }),
+
+      // Seamless integration between Rollup and Babel
+      rollupBabel(
+        Object.assign(
```

Comment:
no spread?

Reply:
Cuz this is just running in Node, so didn't wanna assume Node > 8.X. Ideally folks can run the build on Node 6 as well. See: https://github.com/eventbrite/eventbrite-sdk-javascript/blob/master/.travis.yml#L2-L5